### PR TITLE
Fixed a few warnings that the new compiler (version 7.2.0) detects

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed a few warnings that the new compiler (version 7.2.0) detects

### CODE REVIEW

File: src/lib/jsonParseV2/parseContextAttributeCompoundValue.cpp [hunk 1]
```diff
@@ -124,13 +125,13 @@ std::string parseContextAttributeCompoundValue
   }
   else if (node->IsArray())
   {
-    int counter  = 0;
+    short counter  = 0;
```
Comment (fgalan) [inline src/lib/jsonParseV2/parseContextAttributeCompoundValue.cpp]: Discussed by skype: revert to int usage and make snprintf buffers larger.

File: CMakeLists.txt [hunk 2]
```diff
@@ -80,6 +80,8 @@ add_definitions(-fPIC)
 #
 IF (${DISTRO} STREQUAL "Ubuntu_17.04")
  set (CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -Werror -Wno-deprecated-declarations -fno-var-tracking-assignments")
+ELSEIF (${DISTRO} STREQUAL "Ubuntu_17.10")
+ set (CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -Werror -Wno-deprecated-declarations -fno-var-tracking-assignments")
```
Comment (fgalan) [inline CMakeLists.txt]: This set if the same than the one for 17.04. Maybe you could generalize boths in the same clause (looking for "Ubuntu_17.*" but in the right CMakeList.txt syntax, of course).

Or at least using an AND for joining both STREQUAL clauses.

Probably there are examples of both in existing CMakeLists.txt files.

**Comment by fgalan** [comment]
LGTM

File: CMakeLists.txt (hunk 2 above)
Comment (kzangeli) [reply CMakeLists.txt]: Fixed in 3cb7510

File: src/lib/jsonParseV2/parseContextAttributeCompoundValue.cpp (hunk 1 above)
Comment (kzangeli) [reply src/lib/jsonParseV2/parseContextAttributeCompoundValue.cpp]: Fixed in 3cb7510

**Pull request merged**